### PR TITLE
Bump golang dependencies

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:
Update golang dependencies to their latest versions.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@haircommander PTAL

File: Makefile [hunk 1]
```diff
@@ -283,8 +283,13 @@ ${SHFMT}:
 ${GO_MOD_OUTDATED}:
 	$(call go-build,./vendor/github.com/psampaz/go-mod-outdated)
 
+ZEITGEIST_VERSION := v0.3.5
```
Comment (jrvaldes) [inline Makefile:286]: I recommend moving this change to a separate commit so that the vendor changes are in a specific commit.

**Review by jrvaldes — COMMENTED** [review]
LGTM. @saschagrunert  Thanks for putting this together. 

Consider adding in the commit message the command used to generate the vendor updates. For example:

```bash
go mod tidy
```

File: Makefile (hunk 1 above)
Comment (saschagrunert) [reply Makefile]: Yep, done.

**Comment by saschagrunert** [comment]
/override ci/prow/e2e-gcp-ovn
/override ci/prow/e2e-aws-ovn

@jrvaldes @haircommander PTAL

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Comment by jrvaldes** [comment]
> /override ci/prow/e2e-gcp-ovn 
> /override ci/prow/e2e-aws-ovn
> 

@saschagrunert  are `e2e-gcp-ovn` and `e2e-aws-ovn` still valid and/or required tests?

**Comment by saschagrunert** [comment]
> > /override ci/prow/e2e-gcp-ovn
> > /override ci/prow/e2e-aws-ovn
> 
> @saschagrunert are `e2e-gcp-ovn` and `e2e-aws-ovn` still valid and/or required tests?

Yes, but they seem to fail independently of this change.
